lib/astarte/client/housekeeping/realms.ex

```diff
@@ -123,6 +125,50 @@ defmodule Astarte.Client.Housekeeping.Realms do
     end
   end
 
+  @doc """
+  Updates a realm.
+
+  ## Examples
+
+    Astarte.Client.Housekeeping.Realms.update(client, realm_name, [jwt_public_key_pem: "..."])
+
+    Astarte.Client.Housekeeping.Realms.update(client, realm_name, [device_registration_limit: 100])
+
+  """
+  def update(%Housekeeping{} = client, realm_name, opts)
+      when is_binary(realm_name) and is_list(opts) do
+    request_path = "realms/#{realm_name}"
+    tesla_client = client.http_client
+    realm_data = %{}
+
+    realm_data =
+      case Keyword.fetch(opts, :jwt_public_key_pem) do
+        {:ok, jwt_public_key_pem} ->
+          Map.merge(realm_data, %{jwt_public_key_pem: jwt_public_key_pem})
+
+        :error ->
+          realm_data
+      end
+
+    realm_data =
+      case Keyword.fetch(opts, :device_registration_limit) do
+        {:ok, device_registration_limit} ->
+          Map.merge(realm_data, %{device_registration_limit: device_registration_limit})
+
+        :error ->
+          realm_data
+      end
```

Comment:
In the future, other options for realm update will be present (e.g. changing replication factor). Maybe extract this in a `maybe_put_realm_update_option/2` function? I think something like this could be nice
```elixir
realm_data
|> maybe_put_realm_update_option(:jwt_public_key_pem)
|> maybe_put_realm_update_option(:device_registration_limit)
```

Reply:
Or even just
```elixir
realm_data =
  [:jwt_public_key_pem, :device_registration_limit]
  |> Enum.reduce(%{}, fn key, acc ->
    case Keyword.fetch(opts, key) do
      {:ok, value} -> Map.put(acc, key, value)
      :error -> acc
    end
  end)
```

Reply:
Thanks, I went with Riccardo's code snippet.